### PR TITLE
epson-201401w: init at 1.0.0

### PR DESCRIPTION
**Pull request by Lunarequest**
###### Motivation for this change

This adds the epson-201401 printer driver which supports the following printers L456, L455, L366, L365, L362, L360, L312, L310, L222, L220, L132, L130

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Lunarequest** [comment]
I rebased to master, it looks like qt is busted not the package

**Comment by veprbl** [comment]
The diff now contains unrelated reverts

**Review by 06kellyjac — CHANGES_REQUESTED** [review]
Please squash `maintainers: fix typo in github username` and `maintainers: add lunarequest` together

Also can you change `epson-201401w init at 1.0.0` (commit and title) to `epson-201401w: init at 1.0.0`

File: pkgs/misc/drivers/epson-201401w/default.nix [hunk 1]
```diff
@@ -0,0 +1,69 @@
+{ lib, stdenv, fetchurl, rpmextract, autoreconfHook, file, libjpeg, cups }:
+
+let
+  version = "1.0.0";
+  filterVersion = "1.0.0";
+in stdenv.mkDerivation {
```
[the remark this answers is not in the document]
Comment (Lunarequest) [reply pkgs/misc/drivers/epson-201401w/default.nix:6]: I believe it does not, I will confirm and switch to stdenvNoCC if it does not.

**Comment by Lunarequest** [comment]
> Please squash `maintainers: fix typo in github username` and `maintainers: add lunarequest` together
> 
> Also can you change `epson-201401w init at 1.0.0` (commit and title) to `epson-201401w: init at 1.0.0`

I can rename the commit however, I can am not sure how to squash  `maintainers: fix typo in github username` and `maintainers: add lunarequest` without squashing   `maintainers: fix typo in github username` into  `epson-201401w init at 1.0.0`

File: pkgs/misc/drivers/epson-201401w/default.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (Lunarequest) [reply pkgs/misc/drivers/epson-201401w/default.nix:6]: this doesn't work, it complains about not finding a `acceptable C compiler` while trying to build the filters

File: pkgs/misc/drivers/epson-201401w/default.nix [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+  pname = "epson-201401w";
```
Comment (SuperSandro2000) [inline pkgs/misc/drivers/epson-201401w/default.nix:8]:
```suggestion
in stdenv.mkDerivation {
  pname = "epson-201401w";
```

File: pkgs/misc/drivers/epson-201401w/default.nix [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  inherit version;
+
+  src = fetchurl {
+    # NOTE: Don't forget to update the webarchive link too!
+    urls = [
+      "https://download3.ebz.epson.net/dsc/f/03/00/03/45/41/92e9c9254f0ee4230a069545ba27ec2858a2c457/epson-inkjet-printer-201401w-1.0.0-1lsb3.2.src.rpm"
+      "https://web.archive.org/web/20200725175832/https://download3.ebz.epson.net/dsc/f/03/00/03/45/41/92e9c9254f0ee4230a069545ba27ec2858a2c457/epson-inkjet-printer-201401w-1.0.0-1lsb3.2.src.rpm"
+    ];
+
+    sha256 = "0c60m1sd59s4sda38dc5nniwa7dh1b0kv1maajr0x9d38gqlyk3x";
```
Comment (SuperSandro2000) [inline pkgs/misc/drivers/epson-201401w/default.nix:18]:
```suggestion
    ];
    sha256 = "0c60m1sd59s4sda38dc5nniwa7dh1b0kv1maajr0x9d38gqlyk3x";
```

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 155464` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>epson-201401w</li>
</ul>
</details>

**Pull request merged**